triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -43,24 +55,60 @@ object TriggerServiceFixture {
       mat: Materializer,
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext): Future[A] = {
+    // Launch a toxiproxy instance . Wait on it to be ready to accept
+    // connections.
+    val toxiExe = BazelRunfiles.rlocation("external/toxiproxy_nix/bin/toxiproxy-cmd")
+    val toxiProxyPort = findFreePort()
+    val toxiCmd = toxiExe + " --port " + toxiProxyPort.toString
+    val toxiProxyProc = Process(toxiCmd).run()
+    var retries = 0
+    var scanning = true
+    while (scanning) {
```

Comment:
Can we use `eventually` here or `com.daml.timer.RetryStrategy`?

Reply:
Good call on using `RetryStrategy`. I'd forgotten about it. There are two `eventually` functions but neither are available here—one is in the ledger API test tool, and one is from ScalaTest.

You can use `RetryStrategy` as follows:

```scala
  val eventually = RetryStrategy.constant(attempts = 3, waitTime = 2.seconds)
  // ...
  println("Waiting for Toxiproxy...")
  eventually { (_, _) =>
    new java.net.Socket("localhost", toxiProxyPort).close()
  }
```

Reply:
Looks like this is what was meant?
```scala
    eventually { (_, _) =>
      for {
        _ <- Future (println("Waiting for Toxiproxy..."))
        channel <- Future (new java.net.Socket("localhost", toxiProxyPort))
      } yield(channel.close())
    }
```

Reply:
Ah, yes, now it's all futures. You can probably get away with:

```scala
  val eventually = RetryStrategy.constant(attempts = 3, waitTime = 2.seconds)
  // ...
  Await.result(eventually { (_, _) =>
    println("Waiting for Toxiproxy...")
    new java.net.Socket("localhost", toxiProxyPort).close()
    Future.unit // or `Future.successful(())`, but why not use the constant?
  }, Duration.Inf)
```